service_container.rst

```diff
@@ -227,131 +220,181 @@ and set it on a ``$logger`` property::
+        $message = $siteUpdateManager->notifyOfSiteUpdate();
+        $this->addFlash('success', $message);
+        // ...
+    }
 
-    use Psr\Log\LoggerInterface;
+Just like before, when you ask for the ``SiteUpdateManager`` service, the container
+will automatically instantiate it for you. By reading the type-hints on the ``__construct()``
+method in that class, it takes care of passing the correct services as arguments.
+All of that is taken care of for you.
 
-    class MessageGenerator
+Manually Wiring Arguments
+-------------------------
+
+There are a few cases when an argument to a service cannot be autowired. For example,
+suppose you want to make the admin email configurable:
+
+.. code-block:: diff
+
+    // src/AppBundle/Updates/SiteUpdateManager.php
+    // ...
+
+    class SiteUpdateManager
     {
-        private $logger;
-        private $loggingEnabled;
+        // ...
+    +    private $adminEmail;
 
-        public function __construct(LoggerInterface $logger, $loggingEnabled)
+    -    public function __construct(MessageGenerator $messageGenerator, \Swift_Mailer $mailer)
+    +    public function __construct(MessageGenerator $messageGenerator, \Swift_Mailer $mailer, $adminEmail)
         {
-            $this->logger = $logger;
-            $this->loggingEnabled = $loggingEnabled;
+            // ...
+    +        $this->adminEmail = $adminEmail;
         }
 
-        public function getHappyMessage()
+        public function notifyOfSiteUpdate()
         {
-            if ($this->loggingEnabled) {
-                $this->logger->info('About to find a happy message!');
-            }
+            // ...
+
+            $message = \Swift_Message::newInstance()
+                // ...
+    -            ->setTo('manager@example.com')
+    +            ->setTo($this->adminEmail)
+                // ...
+            ;
             // ...
         }
     }
 
-The class now has a *second* constructor argument. No problem, just update your
-service config:
+If you make this change and refresh, you'll see an error:
+
+    Cannot autowire service "AppBundle\Updates\SiteUpdateManager": argument "$adminEmail"
+    of method "__construct()" must have a type-hint or be given a value explicitly.
+
+That makes sense! There is no way that the container knows what value you want to
+pass here. No problem! In your configuration, you can explicitly set this argument:
 
 .. configuration-block::
 
     .. code-block:: yaml
 
         # app/config/services.yml
         services:
-            app.message_generator:
-                class:     AppBundle\Service\MessageGenerator
-                arguments: ['@logger', true]
+            # ...
+
+            # same as before
+            AppBundle\:
+                resource: '../../src/AppBundle/{Service,Updates}'
+
+            # explicitly configure the service
+            AppBundle\Updates\SiteUpdateManager:
+                arguments:
+                    $adminEmail: 'manager@example.com'
```

Comment:
I still think that the `$` in the argument name is really strange inside a YAML file ... but I was already told that this "can't" be improved :(